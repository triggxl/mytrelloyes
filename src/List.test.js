import React from 'react';
import ReactDOM from 'react-dom';
import ListComponent from './List';
import renderer from 'react-test-renderer';

describe('List Component', () => {
  //wanting to pick something within ListComponent to add and remove an element to for smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
      ReactDOM.render(<ListComponent className="List-cards" />, div);
      ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ListComponent header = {'test header'}/>)
      .toJSON();
      console.log(tree);
      expect(tree).toMatchSnapshot();  
  });
  //also need to create a snapshot test (what could I test as an example?)
})



