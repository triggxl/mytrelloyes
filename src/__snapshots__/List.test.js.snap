// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-cards"
  >
    <h2>
      test header
    </h2>
  </header>
  <div
    className="List-cards"
  >
     
  </div>
</section>
`;
